File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 09/10/2025, 09:40:15
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        buy = prices[0]
        for i in range(1,len(prices)):
            if prices[i] < buy:
                buy = prices[i]
            elif prices[i] - buy > profit:
                profit = prices[i] - buy
        return profit


        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 03/03/2026, 20:06:55
class Solution {
public:
    int maxProfit(vector<int>& nums) {
        int profit = 0;
        int n = nums.size();
        int minPrice = INT_MAX;
        for(int i =0;i<n; i++)
        {
            if(minPrice > nums[i])
            {
                minPrice = nums[i];
            }
            profit = max(profit,nums[i]-minPrice);
        }
        
         return profit;
    }
};